src/app/file-explorer.js

```diff
@@ -0,0 +1,310 @@
+/* global FileReader, confirm, alert */
+var yo = require('yo-yo')
+var csjs = require('csjs-inject')
+var Treeview = require('ethereum-remix').ui.TreeView
+
+var EventManager = require('../lib/eventManager')
+
+var css = csjs`
+  .fileexplorer       {
+    box-sizing        : border-box;
+  }
+  .folder,
+  .file               {
+    font-size         : 14px;
+  }
+  .hasFocus           {
+    background-color  : #F4F6FF;
+  }
+  .rename             {
+    background-color  : white;
+  }
+  .remove             {
+    align-self        : center;
+    padding-left      : 10px;
+  }
+  .activeMode         {
+    display           : flex;
+    justify-content   : space-between;
+    margin-right      : 10px;
+    padding-right     : 19px;
+  }
+  ul                  {
+    padding           : 0;
+  }
+`
+module.exports = fileExplorer
+
+function fileExplorer (appAPI, files) {
+  var fileEvents = files.event
+  var tv = new Treeview({
+    extractData: function (value, tree, key) {
+      var newValue = {}
+      // var isReadOnly = false
+      var isFile = false
+      Object.keys(value).filter(function keep (x) {
+        if (x === '/content') isFile = true
+        // if (x === '/readOnly') isReadOnly = true
```

Comment:
no they are supported by the `files` object, but the file explorer is not treating them.
I was assuming that a read-only-file might still be removed from the file explorer.
Don't know maybe i add it to issues